che-theia-task-extension/src/browser/che-task-variables-contribution.ts

```diff
@@ -28,33 +29,47 @@ export class ProjectPathVariableContribution implements VariableContribution {
     protected readonly cheWsClient: CheWorkspaceClient;
     @inject(SelectionService)
     protected readonly selectionService: SelectionService;
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
 
     async registerVariables(variables: VariableRegistry): Promise<void> {
 
         variables.registerVariable({
             name: 'current.project.path',
             description: 'The path of the project root folder',
             resolve: async () => {
-                const wsRoot = await this.workspaceService.root;
-                if (!wsRoot || !wsRoot.uri) {
+                let errorMessage = 'Get \'current.project.path\' variable error. ';
+
+                const selection = this.selectionService.selection;
+                if (!Array.isArray(selection) || !selection[0] || !selection[0].fileStat || !selection[0].fileStat.uri) {
+                    errorMessage += 'Any selections are not defined in the project tree.';
+                    await this.messageService.error(errorMessage);
                     return undefined;
                 }
-                const rootWorkspaceUri = new URI(wsRoot.uri);
+                const selectionUri = selection[0]!.fileStat!.uri;
 
-                const selection = this.selectionService.selection;
-                if (!Array.isArray(selection) || !selection[0] || !selection[0].fileStat) {
+                const wsRoot = await this.workspaceService.root;
+                const currentRoot = Array.isArray(wsRoot) ? (<Array<FileStat>>wsRoot).find((root: FileStat) => {
+                    // if multi-root
+                    return selectionUri.startsWith(root.uri);
+                }) : <FileStat>wsRoot;
+                if (!currentRoot || !currentRoot.uri) {
+                    errorMessage += 'Get current workspace root URI error.';
+                    await this.messageService.error(errorMessage);
                     return undefined;
                 }
-                const selectionUri = new URI(selection[0]!.fileStat!.uri);
+                const rootWorkspaceUri = currentRoot.uri;
 
-                const rootWorkspacePath = rootWorkspaceUri.path.toString();
-                const selectionPath = selectionUri.path.toString();
-                if (!selectionPath.startsWith(rootWorkspacePath)) {
+                if (!selectionUri.startsWith(rootWorkspaceUri)) {
+                    errorMessage += 'The selection isn\'t under the current workspace root folder.';
+                    await this.messageService.error(errorMessage);
                     return undefined;
                 }
 
-                const relativeSelectionPath = selectionPath.substr(rootWorkspacePath.length);
+                const relativeSelectionPath = selectionUri.substr(rootWorkspaceUri.length);
                 if (!relativeSelectionPath.length) {
+                    errorMessage += 'The selection is on workspace root folder. Select a project.';
```

Comment:
rephrase it `Workspace root folder is selected and is not a project. Please select a project.`